typing/typecore.ml

```diff
@@ -5089,6 +5139,49 @@ and type_andops env sarg sands expected_ty =
   let let_arg, rev_ands = loop env sarg (List.rev sands) expected_ty in
   let_arg, List.rev rev_ands
 
+(*TODO mbungeroth: Why do I need the type annotation here to make this work?*)
+  and comprehension ~loc ~env ~ty_expected ~container_type 
+                  ~sbody ~(comp_typell : Parsetree.comprehension list list) = 
+
+  let from_to_comprehension ~body_env ~env param slow shigh dir =
+    let low = type_expect env slow
+        (mk_expected ~explanation:For_loop_start_index Predef.type_int) in
+    let high = type_expect env shigh
+        (mk_expected ~explanation:For_loop_stop_index Predef.type_int) in
+    let id, new_env = type_for_loop ~loc ~env:body_env ~param  Predef.type_int in
+    (*let body = type_expect new_env sbody ty_expected in*)
+    id, From_to(param, low, high, dir), new_env
+  in
+  let in_comprehension ~body_env ~env param siter= 
+    let item_ty = newvar() in
+    let iter_ty = instance (container_type item_ty) in
+    let iter = type_expect env siter 
+        (mk_expected ~explanation:InComprehensionArgument iter_ty) in
+    let id, new_env = type_for_loop ~loc ~env:body_env ~param item_ty
+    in
```

Comment:
`in` should be on the previous line here. The two standard ways to indent a `let` are:
```ocaml
let ... = ... in
```
and
```ocaml
let ... =
  ....
in
```
You will see some variations of that in the compiler, but its still best to just use those two.